packages/cms/pages/resources/resource/resource.tsx

```diff
@@ -241,13 +263,48 @@ export const Table: React.FunctionComponent = () => {
             icon: 'trash',
             type: 'icon',
             color: 'danger',
-            onClick: console.log
+
+            onClick: item => {
+              setIsModalVisible(true)
+              setDeleteId([item.id])
+            }
           }
         ]
       }
     ]
   }, [resource])
 
+  const ConfirmDeleteModal = () => {
+    const closeModal = () => setIsModalVisible(false)
+
+    if (isModalVisible) {
+      return (
+        <EuiConfirmModal
+          title={`Do you want to delete the selected item${
+            selectedItems.length > 1 ? 's' : ''
+          }`}
+          onCancel={closeModal}
+          onConfirm={() => {
+            deleteItem()
```

Comment:
Let's make this function asynchronous, and `await delete()` so we delete before closing the modal. And also add a loading state to the delete button while the delete is going on.